qa/integration-tests/src/test/java/io/camunda/zeebe/it/backup/BackupIT.java

```diff
@@ -0,0 +1,115 @@
+import io.camunda.zeebe.gateway.admin.backup.BackupRequestHandler;
+import io.camunda.zeebe.gateway.admin.backup.BackupStatus;
+import io.camunda.zeebe.it.clustering.ClusteringRuleExtension;
+import io.camunda.zeebe.it.util.GrpcClientRule;
+import io.camunda.zeebe.protocol.management.BackupStatusCode;
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.containers.localstack.LocalStackContainer;
+import org.testcontainers.containers.localstack.LocalStackContainer.Service;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+
+@Testcontainers
+class BackupIT {
+  @Container
+  private static final LocalStackContainer S3 =
+      new LocalStackContainer(DockerImageName.parse("localstack/localstack:0.14.5"))
+          .withServices(Service.S3);
+
+  private GrpcClientRule client;
+
+  @RegisterExtension
+  private final ClusteringRuleExtension clusteringRule =
+      new ClusteringRuleExtension(1, 1, 1, this::configureBackupStore);
+
+  private BackupRequestHandler backupRequestHandler;
+
+  private void configureBackupStore(final BrokerCfg config) {
+    final var backupConfig = config.getData().getBackup();
+    backupConfig.setStore(BackupStoreType.S3);
+
+    final var s3Config = backupConfig.getS3();
+    final String bucketName = RandomStringUtils.randomAlphabetic(10).toLowerCase();
+    s3Config.setBucketName(bucketName);
+    s3Config.setEndpoint(S3.getEndpointOverride(Service.S3).toString());
+    s3Config.setRegion(S3.getRegion());
+    s3Config.setAccessKey(S3.getAccessKey());
+    s3Config.setSecretKey(S3.getSecretKey());
+
+    // Create bucket before for storing backups
+    final var s3ClientConfig =
+        io.camunda.zeebe.backup.s3.S3BackupConfig.from(
+            bucketName,
+            s3Config.getEndpoint(),
+            S3.getRegion(),
+            S3.getAccessKey(),
+            S3.getSecretKey());
+    try (final var s3Client = S3BackupStore.buildClient(s3ClientConfig)) {
+      s3Client.createBucket(builder -> builder.bucket(bucketName).build()).join();
+    }
+  }
+
+  @BeforeEach
+  void setup() {
+    client = new GrpcClientRule(clusteringRule.getClient());
+    backupRequestHandler = new BackupRequestHandler(clusteringRule.getGateway().getBrokerClient());
+  }
+
+  @Test
+  void shouldBackup() {
+    // given
+    client.createSingleJob("test");
+
+    final long backupId = 2;
+
+    // when
+    backup(backupId);
+
+    // then
+    Awaitility.await("Backup must be completed.")
+        .timeout(Duration.ofMinutes(2))
+        .ignoreExceptions()
```

Comment:
🔧 what exceptions can occur which we want to ignore? I tried without and didn't hit any, maybe we can remove it to be aware of any unexpected failure of the getStatus call?

Reply:
Here we don't want to fail due to exceptions, because sometimes the request can timeout and we want to to just retry. With `ignoreExceptions`, awaitility will retry. Otherwise, it fails immediately resulting in flaky tests. But it will eventually fail, if nothing good happens.

Reply:
ok, I guess adding the specific timeout exception would be best to not hide potential other failure cases but in case it eventually resolves anyway I think it's acceptable.